### PR TITLE
Updated type hint and extended docstring in make_vec_env and make_atari_env

### PR DESCRIPTION
**Pull request by AlexPasqua**
## Description

According with what discussed in #894:
- The function itself was already working with callables, but it wasn't considerent in the type hint of the function's signature.
- Extended the description of the wrapper_class parameter with a link to a Github issue containing more details on the matter.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
(Note: not sure to call it "bug", but it's still a change in the code that solves an issue, not only in the doc)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

### CODE REVIEW

File: stable_baselines3/common/env_util.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def is_wrapped(env: Type[gym.Env], wrapper_class: Type[gym.Wrapper]) -> bool:
 
 
 def make_vec_env(
-    env_id: Union[str, Type[gym.Env]],
+    env_id: Union[str, Type[gym.Env], Callable[..., gym.Env]],
```
Comment (araffin) [inline stable_baselines3/common/env_util.py:39]: I assume we could also update `make_atari_env` too, no?

Comment (AlexPasqua) [reply stable_baselines3/common/env_util.py:39]: Sure, just did it!

**Comment by qgallouedec** [comment]
`Callable[..., Env]]` includes `Type[Env]` as pointed out in https://github.com/DLR-RM/stable-baselines3/issues/894#issuecomment-1263593280.Thus I would suggest 
```python
env_id: Union[str, Callable[..., gym.Env]],
```

**Comment by AlexPasqua** [comment]
> `Callable[..., Env]]` includes `Type[Env]` as pointed out in [#894 (comment)](https://github.com/DLR-RM/stable-baselines3/issues/894#issuecomment-1263593280) I would suggest
> 
> ```python
> env_id: Union[str, Callable[..., gym.Env]],
> ```

That's true, my bad, fixed it

**Comment by qgallouedec** [comment]
Thank you for contributing!

**Pull request merged**